annofabcli/input_data/put_input_data_with_zip.py

```diff
@@ -0,0 +1,162 @@
+                project_id,
+                job_type=ProjectJobType.GEN_INPUTS,
+                job_access_interval=wait_options.interval,
+                max_job_access=wait_options.max_tries,
+            )
+            if result:
+                logger.info(f"入力データの登録が完了しました。")
+            else:
+                logger.warning(f"入力データの登録に失敗しました。または、{MAX_WAIT_MINUTE}分間待っても、入力データの登録が完了しませんでした。")
+
+    def validate(self, args: argparse.Namespace) -> bool:
+        if args.zip is not None:
+            if not Path(args.zip).exists():
+                print(f"{self.COMMON_MESSAGE} argument --zip: ファイルパスが存在しません。 '{args.zip}'", file=sys.stderr)
+                return False
+
+            if not zipfile.is_zipfile(args.zip):
+                print(f"{self.COMMON_MESSAGE} argument --zip: zipファイルではありません。 '{args.zip}'", file=sys.stderr)
+                return False
+
+        return True
+
+    def main(self):
+        args = self.args
+        if not self.validate(args):
+            sys.exit(COMMAND_LINE_ERROR_STATUS_CODE)
+
+        project_id = args.project_id
+        super().validate_project(project_id, [ProjectMemberRole.OWNER])
+        if args.zip is not None:
+            wait_options = get_wait_options_from_args(get_json_from_args(args.wait_options), DEFAULT_WAIT_OPTIONS)
+            self.put_input_data_from_zip_file(
+                project_id,
+                zip_file=args.zip,
+                input_data_name_prefix=args.input_data_name_prefix,
+                wait=args.wait,
+                wait_options=wait_options,
+            )
+
+        else:
+            print(f"引数が不正です。", file=sys.stderr)
+
+
+def main(args):
+    service = build_annofabapi_resource_and_login(args)
+    facade = AnnofabApiFacade(service)
+    PutInputData(service, facade, args).main()
+
+
+def parse_args(parser: argparse.ArgumentParser):
+    argument_parser = ArgumentParser(parser)
+
+    argument_parser.add_project_id()
+
+    file_group = parser.add_mutually_exclusive_group(required=True)
+
+    file_group.add_argument("--zip", type=Path, help=("入力データとして登録するzipファイルのパスを指定してください。"))
+
+    parser.add_argument(
+        "--input_data_name_prefix",
+        type=str,
+        help="入力データとして登録するzipファイルのinput_data_nameを指定してください。省略した場合、 ``--zip`` のパスになります。",
+    )
+
+    parser.add_argument("--wait", action="store_true", help=("入力データの登録が完了するまで待ちます。"))
+
+    parser.add_argument(
+        "--wait_options",
+        type=str,
+        help="入力データの登録が完了するまで待つ際のオプションをJSON形式で指定してください。 ``--wait`` を指定したときのみ有効なオプションです。"
+        " ``file://`` を先頭に付けるとjsonファイルを指定できます。"
+        'デフォルトは ``{"interval":60, "max_tries":360}`` です。'
+        "``interval`` :完了したかを問い合わせる間隔[秒], "
+        "``max_tires`` :完了したかの問い合わせを最大何回行うか。",
+    )
+
+    parser.set_defaults(subcommand_func=main)
+
+
+def add_parser(subparsers: Optional[argparse._SubParsersAction] = None):
+    subcommand_name = "put_with_zip"
+    subcommand_help = "入力データを登録します。"
```

Comment:
```suggestion
    subcommand_help = "zipファイルを入力データとして登録します。"
```

`input_data put`コマンドと説明が重複するので、違いが分かる説明にしました。

Reply:
修正しました。